src/cpp/rtps/xmlparser/XMLParser.cpp

```diff
@@ -683,7 +681,7 @@ XMLP_ret XMLParser::parseXMLCommonSharedMemTransportData(tinyxml2::XMLElement* p
             {
                 if (XMLP_ret::XML_OK != getXMLUint(p_aux0, &aux, 0))
                     return XMLP_ret::XML_ERROR;
-                transport_descriptor->segment_size(static_cast<uint32_t>(aux), static_cast<uint32_t>(aux));
+                transport_descriptor->segment_size(static_cast<uint32_t>(aux));
```

Comment:
Shall we parse maximum message size somewhere?

Reply:
max_message_size belongs to TransportDescriptorInterface, the parse code is already written.

Reply:
But the parser of the common part casts the descriptor to SocketTransportDescriptor!
I have changed the code in 28fcd66dec006892f3bb5d2d43da36529e7e2964